mgmtsystem/azure.py

```diff
@@ -50,6 +50,7 @@ def __init__(self, **kwargs):
         self.storage_key = kwargs["storage_key"]
         self.subscription_id = kwargs["subscription_id"]
         self.tenant_id = kwargs["tenant_id"]
+        self.region = kwargs["provisioning"]["region_api"]
```

Comment:
I just took over the style used for this provider. I agree that this is not the best way to initialize object. But this PR is devoted to support of two new events. I can change init part of this object a bit later if you wish.